sheetScript.gs

```diff
@@ -51,16 +52,23 @@ function updatePlaylists(sheet) {
     for (var iColumn = reservedTableColumns; iColumn < sheet.getLastColumn(); iColumn++) {
       var channel = data[iRow][iColumn];
       if (!channel) continue;
-      else if (channel == "ALL")
-        channelIds.push.apply(channelIds, getAllChannelIds());
-      else if (channel.substring(0,2) == "PL" && channel.length > 10)  // Add videos from playlist. MaybeTODO: better validation, since might interpret a channel with a name "PL..." as a playlist ID
+      else if (channel == "ALL") {
+        var newChannelIds = getAllChannelIds();
+        if (!newChannelIds || newChannelIds.length === 0) Logger.log("Could not find any subscriptions")
+        else [].push.apply(channelIds, newChannelIds);
+      } else if (channel.substring(0,2) == "PL" && channel.length > 10)  // Add videos from playlist. MaybeTODO: better validation, since might interpret a channel with a name "PL..." as a playlist ID
          playlistIds.push(channel);
       else if (!(channel.substring(0,2) == "UC" && channel.length > 10)) // Check if it is not a channel ID (therefore a username). MaybeTODO: do a better validation, since might interpret a channel with a name "UC..." as a channel ID
       {
         try {
-          channelIds.push(YouTube.Channels.list('id', {forUsername: channel, maxResults: 1}).items[0].id);
+          var channel = YouTube.Channels.list('id', {forUsername: channel, maxResults: 1});
+          if (!channel || !channel.items) Logger.log("Cannot query for channel")
+          else if (channel.items.length === 0) Logger.log("No channel with this name")
+          else if (channel.items.length !== 1) Logger.log("Multiple channels with this name")
+          else if (!channel.items[0].id) Logger.log("Cannot get id from this channel")
+          else channelIds.push(channel.items[0].id);
```

Comment:
Could return `+ channel` in the error to make debugging for users easier (Typos etc).

Reply:
Addressed in #48

Reply:
Thanks

Reply:
Merged